models/facial_expression_recognition/demo.cpp

```diff
@@ -0,0 +1,304 @@
+                                                                                        "2: CUDA + GPU (CUDA FP16),\n"
+                                                                                        "3: TIM-VX + NPU,\n"
+                                                                                        "4: CANN + NPU}"
+"{ save s           | false                                                             | Specify to save results.}"
+"{ vis v            | true                                                              | Specify to open a window for result visualization.}"
+;
+
+
+int main(int argc, char** argv)
+{
+    CommandLineParser parser(argc, argv, keys);
+ 
+    parser.about("Facial Expression Recognition");
+    if (parser.has("help"))
+    {
+        parser.printMessage();
+        return 0;
+    }
+    
+    string modelPath = parser.get<string>("model");
+    string yunetModelPath = parser.get<string>("yunet_model");
+    string inputPath = parser.get<string>("input");
+    uint8_t backendTarget = parser.get<uint8_t>("backend_target");
+    bool saveFlag = parser.get<bool>("save");
+    bool visFlag = parser.get<bool>("vis");
+
+    if (modelPath.empty())
+        CV_Error(Error::StsError, "Model file " + modelPath + " not found");
+
+    if (yunetModelPath.empty())
+        CV_Error(Error::StsError, "Face Detection Model file " + yunetModelPath + " not found");
+
+    YuNet faceDetectionModel(yunetModelPath);
+    FER expressionRecognitionModel(modelPath, backend_target_pairs[backendTarget].first, backend_target_pairs[backendTarget].second);
+
+    VideoCapture cap;
+    if (!inputPath.empty())
+        cap.open(samples::findFile(inputPath));
+    else
+        cap.open(0);
+    
+    if (!cap.isOpened())
+        CV_Error(Error::StsError, "Cannot opend video or file");    
+
+    Mat frame;
+    static const std::string kWinName = "Facial Expression Demo";
+
+
+    while (waitKey(1) < 0)
+    {
+        cap >> frame;
+
+        if (frame.empty())
+        {
+            if(inputPath.empty())
+                cout << "Frame is empty" << endl;
+            break;
+        }
+
+        faceDetectionModel.setInputSize(frame.size());
+        
+        Mat faces = faceDetectionModel.infer(frame);
+        vector<String> expressions;
+
+        for (int i = 0; i < faces.rows; ++i)
+        {
+            Mat face = faces.row(i);
+            String exp = expressionRecognitionModel.infer(frame, face);
+            expressions.push_back(exp);
+
+            int x1 = static_cast<int>(faces.at<float>(i, 0));
+            int y1 = static_cast<int>(faces.at<float>(i, 1));
+            int w = static_cast<int>(faces.at<float>(i, 2));
+            int h = static_cast<int>(faces.at<float>(i, 3));
+            float conf = faces.at<float>(i, 14);
+
+            std::cout << cv::format("%d: x1=%d, y1=%d, w=%d, h=%d, conf=%.4f expression=%s\n", i, x1, y1, w, h, conf, exp.c_str());
```

Comment:
I believe this is a C++20 feature. We use C++11 as same as OpenCV 4.x does. Please also add

```cmake
set(CMAKE_CXX_STANDARD 11)
```

in the CMake script.

Reply:
Do you mean the `format`?